solr/core/src/java/org/apache/solr/schema/IndexSchemaFactory.java

```diff
@@ -62,7 +62,7 @@ public static IndexSchema buildIndexSchema(String resourceName, SolrConfig confi
     PluginInfo info = config.getPluginInfo(IndexSchemaFactory.class.getName());
     IndexSchemaFactory factory;
     if (null != info) {
-      factory = config.getResourceLoader().newInstance(info.className, IndexSchemaFactory.class);
+      factory = config.getResourceLoader().newInstance(info, IndexSchemaFactory.class);
```

Comment:
Do we even support the `packageName:ClassName` scheme in `schema.xml` at all? How does it play with this?
Need more discussions

Reply:
> Do we even support the packageName:ClassName scheme in schema.xml at all?

It does!  Woohoo!  It's because all loading occurs via a SolrResourceLoader in Solr and so that's where the focal point of the approach in this PR is.

Reply:
That's plain wrong. SRL has no idea which version of package should be loaded

Reply:
Sound like an exciting design, SRL in response of loading stuff. Could not SRL ask SolrCore about version details if it is a core-level package?

Reply:
"That's plain wrong. "  Please work on your tone @noblepaul .  There is no right/wrong here; we are exploring what the future holds together; it has not been written yet with no spec to judge right/wrong.  It would be more correct to say "According to my vision, the SRL should not know about packages".

Reply:
> There is no right/wrong here; 

This is not a personal remark. What I'm saying is "it's plain wrong to load using the latest version". 

This is not how the rest of the system work. So, we have to ensure that what should be the expectation of the user

Reply:
> am I supposed to make comments on what it does today?

Yes (agreed); and not the future.

> I never said SRL loading classes is "plain wrong

I'm glad to hear that.  You made this statement immediately after what I said, which was "It does! Woohoo! It's because all loading occurs via a SolrResourceLoader in Solr and so that's where the focal point of the approach in this PR is."  So I thought you meant my most recent statement "was plain wrong" vs some statement in some other comment thread (notice this thread doesn't discuss versions).  Can you see how I came to this interpretation?

I am working this week, albeit lightly BTW.